src/ensembl/src/content/app/species-selector/components/popular-species-button/PopularSpeciesButton.tsx

```diff
@@ -48,19 +48,24 @@ export const PopularSpeciesButton = (props: Props) => {
 
   const handleClick = () => {
     const { genome_id, is_available } = species;
+    const speciesName = species.common_name || species.scientific_name;
+
     if (!is_available) {
       return;
     } else if (isSelected) {
       props.clearSelectedSpecies();
+      analyticsTracking.trackEvent({
+        category: 'popular_species',
+        action: 'unpreselect',
+        label: speciesName
```

Comment:
Let's move this out of the body of the component function into a separate function (`sendAnalytics`?) that takes action (preselect/unpreselect) and speciesName and sends a track event?

Reply:
How would creating a separate function be useful than directly calling `analyticsTracking.trackEvent`?

Reply:
There are two calls to it in the body of the component function: one in the else if branch, another in the else branch. Moving this bit of code out of the body of the component function (I don't mean to a different file) will tighten up the code slightly.

Reply:
I'm not sure if it's a good idea to create another wrapper for the analytics just to reduce 2 lines of code?

Reply:
🤷‍♂ 

ok, let's leave it as it is then.